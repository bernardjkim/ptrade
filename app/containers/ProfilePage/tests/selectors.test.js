import { fromJS } from 'immutable';
import {
  selectProfilePageDomain,
  makeSelectError,
  makeSelectLoading,
  makeSelectPortfolioValue,
  makeSelectTimeFrame,
} from '../selectors';

const mockedState = fromJS({
  profilePage: {
    loading: false,
    error: 404,
    profileValue: false,
    timeFrame: 0,
  },
});

describe('selectProfilePageDomain', () => {
  it('Expect to have unit tests specified', () => {
    expect(selectProfilePageDomain).toMatchSnapshot();
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading', () => {
    expect(loadingSelector(mockedState)).toMatchSnapshot();
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error', () => {
    expect(errorSelector(mockedState)).toMatchSnapshot();
  });
});

describe('makeSelectPortfolioValue', () => {
  const portfolioValueSelector = makeSelectPortfolioValue();
  it('should select the error', () => {
    expect(portfolioValueSelector(mockedState)).toMatchSnapshot();
  });
});

describe('makeSelectTimeFrame', () => {
  const timeFrameSelector = makeSelectTimeFrame();
  it('should select the error', () => {
    expect(timeFrameSelector(mockedState)).toMatchSnapshot();
  });
});
