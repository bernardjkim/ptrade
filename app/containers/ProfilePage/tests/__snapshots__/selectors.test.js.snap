// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectError should select the error 1`] = `404`;

exports[`makeSelectLoading should select the loading 1`] = `false`;

exports[`makeSelectPortfolioValue should select the error 1`] = `undefined`;

exports[`makeSelectTimeFrame should select the error 1`] = `0`;

exports[`selectProfilePageDomain Expect to have unit tests specified 1`] = `[Function]`;
